distribution/pull_v2.go

```diff
@@ -32,7 +32,10 @@ import (
 	"golang.org/x/net/context"
 )
 
-var errRootFSMismatch = errors.New("layers from manifest don't match image configuration")
+var (
+	errRootFSMismatch  = errors.New("layers from manifest don't match image configuration")
+	errMediaTypePlugin = errors.New("plugins cannot be pulled in this manner. Use `docker plugin install`")
```

Comment:
bit unsure about "cannot be pulled in this manner", perhaps 

    invalid manifest type for image. Use `docker plugin install`

but not really satisfied with that one @sfsmithcha perhaps you have a suggestion?